service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/create/CreateAzureDiskStep.java

```diff
@@ -0,0 +1,103 @@
+package bio.terra.workspace.service.resource.controlled.flight.create;
+
+import bio.terra.cloudres.azure.resourcemanager.common.Defaults;
+import bio.terra.cloudres.azure.resourcemanager.compute.data.CreateDiskRequestData;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.app.configuration.external.AzureConfiguration;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.resource.controlled.ControlledAzureDiskResource;
+import bio.terra.workspace.service.workspace.model.AzureCloudContext;
+import com.azure.core.management.Region;
+import com.azure.core.management.exception.ManagementException;
+import com.azure.resourcemanager.compute.ComputeManager;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Creates an Azure Disk. Designed to run directly after {@link GetAzureDiskStep}. */
+public class CreateAzureDiskStep implements Step {
+
+  private static final Logger logger = LoggerFactory.getLogger(CreateAzureDiskStep.class);
+  private final AzureConfiguration azureConfig;
+  private final CrlService crlService;
+  private final ControlledAzureDiskResource resource;
+  private final AzureCloudContext azureCloudContext;
+
+  public CreateAzureDiskStep(
+      AzureConfiguration azureConfig,
+      AzureCloudContext azureCloudContext,
+      CrlService crlService,
+      ControlledAzureDiskResource resource) {
+    this.azureConfig = azureConfig;
+    this.azureCloudContext = azureCloudContext;
+    this.crlService = crlService;
+    this.resource = resource;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    ComputeManager computeManager = crlService.getComputeManager(azureCloudContext, azureConfig);
+
+    try {
+      computeManager
+          .disks()
+          .define(resource.getDiskName())
+          .withRegion(resource.getRegion())
+          .withExistingResourceGroup(azureCloudContext.getAzureResourceGroupId())
+          .withData()
+          .withSizeInGB(resource.getSize())
+          .withTag("workspaceId", resource.getWorkspaceId().toString())
+          .withTag("resourceId", resource.getResourceId().toString())
+          .create(
+              Defaults.buildContext(
+                  CreateDiskRequestData.builder()
+                      .setName(resource.getDiskName())
+                      .setRegion(Region.fromName(resource.getRegion()))
+                      .setResourceGroupName(azureCloudContext.getAzureResourceGroupId())
+                      .setSize(resource.getSize())
```

Comment:
After updating to CRL 0.4.0, add:
```
  .setTenantId(azureCloudContext.getAzureTenantId())
  .setSubscriptionId(azureCloudContext.getAzureSubscriptionId())
```
to all Azure `ComputeManager` calls.

This should allow Janitor to receive messages for CRL-created resources.